### ISSUE TITLE
autotools-based build system?

### ISSUE THREAD

**Issue by edwardhartnett** [lz4/lz4#916]
I am working on a project that hopes to bring LZ4 compression to the netCDF community. NetCDF is a scientific data format widely used in meteorology and climate science (see https://en.wikipedia.org/wiki/NetCDF).

Currently netCDF offers only zlib compression for science data, we hope to change that with the CCR project: https://github.com/ccr/ccr.

Our users will have to install LZ4 on their platform, usually a Linux workstation or High Performance Computer (HPC, a.k.a. supercomputers).

Our users are used to standard autotools builds, with a configure step, followed by make.

Is there a plan for an autotools based build system for LZ4? Would you accept a contributed solution?

Thanks!

**Comment by Cyan4973** [comment]
Hi @edwardhartnett 

There is currently no plan to develop an `autotools` based build system.
`lz4` offers just the last stage, `make`,
which has so far proved enough for the vast majority of Linux and Posix environments,
likely thanks to the fact that `lz4` has very few dependencies.

Additionally, an optional `cmake` script is provided in `build/cmake`.

It's possible to host other build scripts and system, such as `autotools`.
It would typically start its life within `contrib/`, to emphasize the fact that it's community-maintained.

**Comment by edwardhartnett** [comment]
OK, not sure that's possible, autoconf expects to have the top-level directory as its home.

**Comment by kwesolowski** [comment]
Why would anyone care for configure step if make step is sufficient? @edwardhartnett what are the build failures you experienced?

**Comment by edwardhartnett** [comment]
@kwesolowski this would need to be installed on many HPC and other operational systems. A standard build system would be required. No one wants to help debug your makefiles. ;-) Specifically we expect standard configure arguments (ex. prefix, disable-shared, etc.) to work, and the config.log file provides a great (and standard) way to debug build issues. CMake is acceptable, but more frequently causes problems than an autotools build.

In fact, we did not use LZ4. Something is apparently wrong with the HDF5 filter code that uses LZ4. Whenever I produced a file with it, the file was actually larger than the original, not smaller. Details here: https://github.com/ccr/ccr/issues/20.

Unfortunately, I don't have time to try and fix the HDF5 filter code, so we just bypassed LZ4. If the filter gets repaired, I'll try LZ4 again. (I realize the filter is not part of LZ4, but it's not something I wrote either.)
